### PR TITLE
Update Supported Kubernetes to 1.26, 1.27, 1.28

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>

/kind breaking

> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Work on #3371 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/gameserverallocations/allocator_test.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ func TestAllocatorAllocate(t *testing.T) {
 
 	require.NoError(t, a.Run(ctx))
 	// wait for it to be up and running
+	// nolint:staticcheck
 	err := wait.PollImmediate(time.Second, 10*time.Second, func() (done bool, err error) {
```
Comment (gongmax) [inline pkg/gameserverallocations/allocator_test.go:85]: I saw you fixed this lint warning with `PollUntilContextTimeout` which looks correct. Do you mind fix these too? We can also do this in a separate PR. Up to you.

File: pkg/gameserverallocations/allocator.go [hunk 2]
```diff
@@ -692,6 +692,7 @@ func Retry(backoff wait.Backoff, fn func() error) error {
 			return false, nil
 		}
 	})
+	// nolint:staticcheck
```
Comment (gongmax) [inline pkg/gameserverallocations/allocator.go:695]: Let's create an issue to track the fix.

**Review by gongmax — COMMENTED** [review]
Did you perform step "Regenerate Kubernetes resource includes (e.g. ObjectMeta, PodTemplateSpec)"? I didn't see any change for ObjectMeta and PodTemplateSpec.

**Comment by Kalaiselvi84** [comment]
> Regenerate Kubernetes resource includes

Yes, I've executed this step, and within this PR, neither _io.k8s.api.core.v1.PodTemplateSpec.yaml nor _io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta.yaml were modified.
These two files were previously altered by the PR related to the [Go script conversion](https://github.com/googleforgames/agones/pull/3413/files#diff-b40ff7394c6e4e3d7105988c2dae58b41a6b4872b636b153d4c9e9ed99fd3757). For reference, I set up the 1.27 cluster before this conversion took place.

File: pkg/gameserverallocations/allocator_test.go (hunk 1 above)
Comment (Kalaiselvi84) [reply pkg/gameserverallocations/allocator_test.go:85]: sure, will replace all `PollImmediate` with `PollUntilContextTimeout`.

Comment (Kalaiselvi84) [reply pkg/gameserverallocations/allocator_test.go]: I've replaced `PollImmediate` with `PollUntilContextTimeout`.
Is it fine to replace `wait.Poll(time.Second, 10*time.Second, func() (bool, error)` with `wait.PollUntilContextTimeout(ctx, time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error))`?

Comment (gongmax) [reply pkg/gameserverallocations/allocator_test.go]: Yes, `wait.Poll` can be replaced with `wait.PollUntilContextTimeout`. Again, up to you whether do that in this PR or a following up PR.

File: pkg/sdkserver/sdkserver.go [hunk 3]
```diff
@@ -295,6 +295,7 @@ func (s *SDKServer) WaitForConnection(ctx context.Context) error {
 	}
 
 	try := 0
+	// nolint:staticcheck
 	return wait.PollImmediateInfiniteWithContext(ctx, 4*time.Second, func(ctx context.Context) (bool, error) {
```
Comment (gongmax) [inline pkg/sdkserver/sdkserver.go:299]: This can also be replaced with `PollUntilContextCancel`

File: test/e2e/framework/framework.go [hunk 4]
```diff
@@ -292,6 +292,7 @@ func (f *Framework) WaitForGameServerState(t *testing.T, gs *agonesv1.GameServer
 // Each Allocated GameServer gets deleted allocDuration after it was Allocated.
 // GameServers will continue to be Allocated until a message is passed to the done channel.
 func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
+	// nolint:staticcheck
 	err := wait.PollImmediateUntil(period, func() (bool, error) {
```
Comment (gongmax) [inline test/e2e/framework/framework.go:296]: This can also be replaced with [PollUntilContextCancel](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediateUntil)

**Review by gongmax — APPROVED** [review]
There are some deprecated methods can be replaced. But I'm totally fine to fix it in a following PR if you prefer.

**Comment by Kalaiselvi84** [comment]
> There are some deprecated methods can be replaced. But I'm totally fine to fix it in a following PR if you prefer.

Will create a separate PR for this fix😅
